tiledb/sm/misc/comparators.h

```diff
@@ -102,22 +98,26 @@ class ColCmp {
    * @param b The second coordinate.
    * @return `true` if `a` precedes `b` and `false` otherwise.
    */
-  bool operator()(const ResultCoords<T>& a, const ResultCoords<T>& b) const {
-    for (unsigned int i = dim_num_ - 1;; --i) {
-      if (a.coords_[i] < b.coords_[i])
+  bool operator()(const ResultCoords& a, const ResultCoords& b) const {
+    for (unsigned int d = dim_num_ - 1;; --d) {
+      auto res = domain_->cell_order_cmp(d, a.coord(d), b.coord(d));
+
+      if (res == -1)
         return true;
-      if (a.coords_[i] > b.coords_[i])
+      if (res == 1)
         return false;
-      // else a.coords_[i] == b.coords_[i] --> continue
+      // else same coordinate on dimension d --> continue
 
-      if (i == 0)
+      if (d == 0)
```

Comment:
let's put this in the for-loop on line 102: `for (unsigned int d = dim_num_ - 1; d >= 0; --d) {`

Reply:
The `d>=0` will not take any effect here. When `d` becomes `<0`, it will wrap around as it is unsigned. I can change the variable from `i` to `d` everywhere though, as I do so in other places of the code and I like it better :).

Reply:
Ah, you're right. Thanks for the clarification.